app/controllers/favorites_controller.rb

```diff
@@ -24,7 +24,10 @@ def create
 
   def destroy
     favorite = Favorite.find_by(id: params[:id])
-    favorite.destroy
+    if favorite.user_id == current_user.id
```

Comment:
A more rails-like way to manage this is:

```ruby
favorite = current_user.favorites.find(params[:id])
```
It scopes the favorites to only those that are for the current_user.